File: Bucket and Water Flow.c
#include<stdio.h>
int main()
{
    int t;
    scanf("%d",&t);
    while(t--)
    {
        int x,z,y,w;
        scanf("%d%d%d%d",&w,&x,&y,&z);
        if(w+y*z==x)
        printf("filled\n");
        else if(w+y*z<x)
        printf("unfilled\n");
        else
        printf("overfilled\n");
    }
}

File: Bucket and Water Flow.java
import java.util.*;
class Codechef
{
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t-- >0)
        {
            int w=sc.nextInt();
            int x=sc.nextInt();
            int y=sc.nextInt();
            int z=sc.nextInt();
            if(x-w>y*z)
            System.out.println("Unfilled");
            else if(x-w==y*z)
            System.out.println("filled");
            else
            System.out.println("overFlow");
        }
    }
}